core/include/prometheus/gauge_builder.h

```diff
@@ -14,20 +14,61 @@ namespace detail {
 class GaugeBuilder;
 }
 
+/// \brief Return a builder to allow configuration of a family of gauges.
+///
+/// A family groups a set of gauges under a common name tagged with labels.
+///
+/// Example usage:
+///
+/// \code
+/// auto registry = std::make_shared<Registry>();
+/// auto& counter_family = prometheus::BuildGauge()
+///                            .Name("some_name")
+///                            .Help("Additional description.")
+///                            .Labels({{"key", "value"}})
+///                            .Register(*registry);
+///
+/// ...
+/// \endcode
+///
+/// Every time series is uniquely identified by its metric name and a set of
+/// key-value pairs, also known as labels. Use GaugeBuilder::Name(const
+/// std::string&) to set the metric name. Use GaugeBuilder::Label(const
+/// std::map<std::string, std::string>&) to set a label. Use
+/// GaugeBuilder::Help(const std::string&) to set an additional description.
+///
+/// It is required to follow the syntax of metric names and labels given by:
+/// https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels
+///
+/// The following metric and label conventions are not required for using
+/// Prometheus, but can serve as both a style-guide and a collection of best
+/// practices: https://prometheus.io/docs/practices/naming/
+///
+/// Finally the gauge family must be registered as a collector with
+/// GaugeBuilder::Register(Registry&). The gauge family fullfills the
+/// Collectable interface to expose the collected metrics. The registration is a
+/// key part needed to scrape metrics.
 detail::GaugeBuilder BuildGauge();
 
 namespace detail {
 class GaugeBuilder {
  public:
+  /// \brief Configure a set of key-value pairs (= labels) of the metric.
```

Comment:
See my comment above. It is a little bit unfortunate to document a class in namespace detail for a user.